common/log/platform/boost/boost_log.cpp

```diff
@@ -184,6 +187,31 @@ void SetLevel(LogLevel level) {
 	global_logger_.SetLevel(level);
 }
 
+error::Error SetupFileLogging(const string &log_file_path, bool exclusive) {
+	typedef sinks::synchronous_sink<sinks::text_ostream_backend> text_sink;
+	boost::shared_ptr<text_sink> sink = boost::make_shared<text_sink>();
+
+	// Add a stream to write log to
+	auto log_stream = boost::make_shared<std::ofstream>(log_file_path);
+	if (!(log_stream.get())) {
```

Comment:
`get()` only gives me the stream from the `boost::make_shared()` wrapper. It's the `!()` that checks the stream, using the [bool operator](https://en.cppreference.com/w/cpp/io/basic_ios/operator_bool). Or am I missing something?

Reply:
Ah ok, yes, that will also work, I think. But there is a sneaky bug then, because `.get()` returns a raw pointer, so what you're testing is the C-style "is pointer non-null?". To do it correctly you need to write: `(!(*log_stream.get()))`.

I would argue that using `->bad()` is perhaps a bit more explicit and harder to get wrong, but I won't object to the `operator bool` style.

Reply:
I used `!()` because `bool` is a bit more than just `bad()`:
![image](https://user-images.githubusercontent.com/1044278/223681061-b24111c8-44c8-46d6-9d9c-1b93c520755f.png)

Reply:
> Ah ok, yes, that will also work, I think. But there is a sneaky bug then, because .get() returns a raw pointer

Oh, of course. But I was told there will be no pointers in C++! :wink:

Reply:
Hehe, yeah, that one snuck up on you from under a rock. You should quickly bury it again!

Reply:
> I used `!()` because `bool` is a bit more than just `bad()`:

You could use `!good()`, which has the same response as the bool variant.

Reply:
It doesn't, actually, it differs in the way it treats `eof()`.

Reply:
Bah, you're right. So many variants there... But `fail()` is exactly inverse bool, isn't it?

Reply:
I like the use of the `bool` operator (even though I have to say the code is quite cryptic, but that's just C++ syntax to blame).

Reply:
It's just the semantics of the `bool` operator which is something like "is this file stream OK?" and that makes most sense to me. Rather than always going to the docs checking what exactly `fail()`, `bad()`,... mean. But I will, hopefully, change this code soon anyway.